source/extensions/filters/http/common/aws/region_provider.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Common {
+namespace Aws {
+
+class RegionProvider {
```

Comment:
I had a look at how this works in https://github.com/envoyproxy/envoy/pull/5546. This allows both static and environment to specify region. I guess my main question is "do we really need this?". Specifically, Envoy's `Node` ID includes locality information (and a variety of ways to specify, from bootstrap config to CLI). Can we pull region from that in the interest of economy of configuration mechanism and making this more consistent with the rest of Envoy config?

Reply:
My intention was to eventually add a region provider impl that pulls the region from an AWS-endpoint, much like we pull the credentials.

Reply:
I guess the larger point is "shouldn't we be synthesizing the Node locality in the bootstrap or CLI for Envoys based on the AWS endpoint data source?" If you do that, there is no need to manage data sources for this information in the extension.

Credentials can also be placed in Node metadata, but the limitation there is that if they change dynamically during the lifetime of the Envoy, they can become stale. For region, this should be static for the lifetime of an Envoy.

Reply:
Sorry for not addressing that larger point. I think it makes sense. I'll pipe the node metadata through in the later PR.